src/fields.jl

```diff
@@ -52,3 +60,233 @@ Field(LX, data, grid::Grid1D) = Field1D(LX, data, grid)
 Constructs a 2D field of `data` at location `(LX, LY)` on `grid`.
 """
 Field(LX, LY, data, grid::Grid2D) = Field2D(LX, LY, data, grid)
+
+function fill_halos(data, grid::Grid1D{Periodic}, LX::Type{Centre})
+    nx, hx, dx = grid.nx, grid.hx, grid.dx
+    offsetData = zero(grid.xC)
```

Comment:
This implies allocations!

Reply:
And on top of that I don't understand why filling halos requires data input. See the comment below.

Reply:
It should require data input in my opinion. Suppose the user calls Fields and provides some data. Now, it's our responsibility to extend the domain to incorporate halo points, and to fill them up with appropriate values.

Reply:
In the danger of becoming repetitive, I will say again: The operation of filling up the halos does not require any data input. It just requires the knowledge of the boundary conditions so as to fill up the halos in the right manner.

:)

I am arguing that filling up halos does not require data input. You are saying it does. Could you give an example?